prefect_azure/credentials.py

```diff
@@ -388,24 +388,23 @@ class ContainerInstanceCredentials(Block):
     """
     Block used to manage Azure Container Instances authentication. Stores Azure Service
     Principal authentication data.
-
-    Args:
-        client_id: The service principal client ID.
-        tenant_id: The service principal tenant ID.
-        client_secret: The service principal client secret.
-
-    Example:
-        Load stored Azure Blob Storage credentials:
-        ```python
-        from prefect_azure import ContainerInstanceCredentials
-        azure_credentials_block = ContainerInstanceCredentials.load("BLOCK_NAME")
-        ```
     """
 
     _block_type_slug = "container-instance-credentials"
     _block_type_name = "Azure Container Instance Credentials"
     _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/6AiQ6HRIft8TspZH7AfyZg/39fd82bdbb186db85560f688746c8cdd/azure.png?h=250"  # noqa
 
-    client_id: str
-    tenant_id: str
-    client_secret: SecretStr
+    client_id: str = Field(
+        default="...", title="Client ID", description="The service principal client ID."
+    )
+    tenant_id: str = Field(
+        default=..., title="Tenant ID", description="The service principal tenant ID."
+    )
+    client_secret: SecretStr = Field(
+        default=..., description="The service principal client secret."
+    )
+    subscription_id: SecretStr = Field(
```

Comment:
The only downside to putting the subscription in the credentials block is that it ties subscription to identity even though Azure treats them as independent.

A single service principal can spin up resources for multiple subscriptions. So for example, you might have dev, staging, and production subscriptions and want to use the same service principal for creating containers in each of them. 

Overall though, I *think* it's enough of an edge case that it doesn't matter too much either way. I had left `subscription_id` in the container instance block instead of putting it in the credentials block to match the way the Azure APIs use subscription IDs. 

But some parts of the Azure APIs feel cobbled-together and unintuitive, so we should do whatever will make the most sense to Prefect users - and from that perspective, I think having subscription ID as part of the credentials block is the right call.

Reply:
Oops, I think I moved it back out after reading more about it, but I forgot to remove it from here.

Reply:
Nevermind, the latest changes does revert the subscription id back into the container instance job.

Reply:
Ah - I missed that in the `fix tests` commit; apologies for the long comment about it. Sounds like we're on the same page here. I initially put the subscription in the credentials block as well. 😄

Reply:
No worries, I had a sloppy commit message for that